lib/calendar.ts

```diff
@@ -194,7 +194,7 @@ export class Calendar implements Temporal.Calendar {
   fields(fields: Params['fields'][0]): Return['fields'] {
     if (!ES.IsTemporalCalendar(this)) throw new TypeError('invalid receiver');
     const fieldsArray = [] as string[];
-    const allowed = new Set(['year', 'month', 'monthCode', 'day']);
+    const allowed = new OriginalSet(['year', 'month', 'monthCode', 'day']);
```

Comment:
This commit looks like it mostly collapsed with some other commit during the rebase. Should the `Set` → `OriginalSet` here be absorbed into that commit as well?

Reply:
This was https://github.com/tc39/proposal-temporal/commit/09c2c824c93edd135e50ec5c3d0eecef9a0a5932. 

I was skeptical about porting over the rest of that commit because AFAICT it wasn't doing anything except manually implementing `for...of` in a precise but confusing (and slow-performing) way. 

Do you see any other changes from that commit that are needed to port?

FWIW, in general in the production polyfill, we're only dealing with caching primordials to make it easier to sync with changes to upstream. Once upstream changes have slowed, I suspect we'll probably want to revert all those cached primordials and just use built-ins. (For bundle size, performance, and maintainability.)